lib/beaker-hostgenerator/generator.rb

```diff
@@ -77,34 +77,21 @@ def generate(tokens, options)
       return config.to_yaml
     end
 
-    def __get_role_config(role)
-      begin
-        r = BeakerHostGenerator::Roles.new
-        m = r.method(role)
-      rescue NameError
-        return {}
-      end
-
-      return m.call
-    end
-
-    def __parse_node_info_token(token)
-      node_info = NODE_REGEX.match(token)
-
-      if node_info
-        node_info = Hash[ node_info.names.zip( node_info.captures ) ]
+    def __generate_host_roles!(host_config, node_info, options)
+      if not options[:disable_default_role]
+        host_config['roles'] = ['agent']
```

Comment:
I wouldn't say this is action at a distance since this is a private function that lives below the only other function it's used in. It's a completely encapsulated function that sets up the host role portion of the data structure.

I suppose we could take the approach of:
```ruby
host_config = __generate_host_roles(host_config, node_info, options)
```
But then `__generate_host_roles` would have to clone the object.

Given that these are all private functions that are meant to decompose the larger `generate` method, I think we can take the liberty of writing the private functions in a way that is most convenient for the Generator and the `generate` method.

I actually was thinking that we'd end up with several functions of the form `configure_something_about_the_host_config!(host_config, ...)`, and then the `generate` method would just call them like so:
```ruby
def generate(...)
  host_config = base_host_config(options)
  configure_X!(host_config, options)
  configure_Y!(host_config, options)
  configure_Z!(host_config, options)
  host_config
end
```